### PR TITLE
[REEF-384]  Avoid Null pointer exception in AvroConfigurationSerializer

### PR DESCRIPTION
**Pull request by jwang98052**
This PR added null checking in AvroConfigurationSerializer.FromAvro() and added tests cases for it.
It also fixed a Tang test case

JIRA: REEF-384(https://issues.apache.org/jira/browse/REEF-384)

This closes #

Author: Julia Wang  Email: jwang98052@yahoo.com

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Tang.Tests/ClassHierarchy/TestClassHierarchy.cs [hunk 1]
```diff
@@ -189,10 +189,11 @@ public void TestNamedRepeatConstructorArgClasses()
         }
 
         [TestMethod]
-        public void TestResolveDependencies() 
+        public void TestResolveDependencies()
         {
-            ns.GetNode(typeof(SimpleConstructors).AssemblyQualifiedName);
-            Assert.IsNotNull(ns.GetNode(typeof(string).AssemblyQualifiedName));
+            ns = TangFactory.GetTang().GetClassHierarchy(new string[] { FileNames.Examples });
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tang.Tests/ClassHierarchy/TestClassHierarchy.cs]: You shouldn't need the `string[]` here, as the method takes / should take var args.

File: lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestAvroConfiguration.cs [hunk 2]
```diff
@@ -32,25 +32,52 @@ namespace Org.Apache.REEF.Tang.Tests.Configuration
...
+
+            var s = (AvroConfigurationSerializer)_serializer;
+            var c = s.FromAvro(s.ToAvroConfiguration(conf1));
+            Assert.IsNotNull(c);
+        }
+
+        [TestMethod]
+        public void TestAddFromAvroWithNull()
+        {
+            var c = ((AvroConfigurationSerializer)_serializer).FromAvro(null);
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestAvroConfiguration.cs]: Why not leave the type of `_serializer` to be `AvroConfigurationSerializer`? The change only seems to complicate matters.

File: lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestAvroConfiguration.cs [hunk 3]
```diff
@@ -32,25 +32,52 @@ namespace Org.Apache.REEF.Tang.Tests.Configuration
...
+            Assert.IsNull(c);
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tang.Tests/Configuration/TestAvroConfiguration.cs]: Is this a common idiom on C#? I'd expect this to throw a `NullPointerException` instead. Passing `null` here seems to indicate a bug in the user code, right?

**Comment by markusweimer** [comment]
I did a pass. I am not sure this needs fixing. Passing `null` to a method that is meant to deserialize what's passed to it is OK to throw an exception. Ideally one which indicates exactly what went wrong. What's the rationale for adding support for passing `null`?

File: lang/cs/Org.Apache.REEF.Tang.Tests/ClassHierarchy/TestClassHierarchy.cs (hunk 1 above)
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tang.Tests/ClassHierarchy/TestClassHierarchy.cs]: The API uses array. We can change it in a separate Jira as it involves more changes.

**Comment by jwang98052** [comment]
I am fine to leave the responsibility to the caller.

File: lang/cs/Org.Apache.REEF.Tang.Tests/ClassHierarchy/TestClassHierarchy.cs (hunk 1 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tang.Tests/ClassHierarchy/TestClassHierarchy.cs]: I filled [REEF-399](https://issues.apache.org/jira/browse/REEF-399) for that.

**Comment by markusweimer** [comment]
Let's move the discussion of a good `null` policy for .NET to the mailing list and then revisit.

**Pull request closed without merging**